src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Remoting.Messaging;
+using System.Web.Mvc;
+using HealthCheck.Core;
+using HealthCheck.Mvc.Models;
+
+namespace HealthCheck.Mvc.Controllers
+{
+    public class MvcHealthCheckController : AsyncController
+    {
+        private const string Key = "result";
+        private readonly Core.HealthCheck _healthCheck;
+
+        public MvcHealthCheckController()
+        {
+            _healthCheck = new Core.HealthCheck(MvcHealthCheckСonfigurator.GetInstance().GetCheckers());
+        }
+
+        public void RunAsync()
+        {
+            AsyncManager.OutstandingOperations.Increment();
+            AsyncManager.Parameters[Key] = _healthCheck.Run().Result;
+            AsyncManager.OutstandingOperations.Decrement();
+        }
+
+        public ActionResult RunCompleted(HealthCheckResult result)
+        {            
+            var model = new HealthCheckResultModel(result);
+            return View("~/Views/HealthCheck.cshtml", model);
```

Comment:
Use the built-in content negotiation to return the requested view instead of always forcing the response to be HTML. So just use `return View(model);` - I'm not sure if the view locator would need some additional hint to find the correct HTML view in this case but I think it will work out of the box.

This way, users can get a JSON healthcheck endpoint in an ASP.NET MVC project just by setting the `Accept` header when requesting it.

Reply:
But that is the point. Healthcheck page is something different then api call.

Reply:
I have already added additional virtual path provider to discover that view. Dont remember exactly but it was not easy or even possible to use what are you talking about.